### ISSUE TITLE
Game client segfaults when attempting to update add-on.

### ISSUE THREAD

**Issue by knyghtmare** [wesnoth/wesnoth#5917]
### Game and System Information
 - What version of the game are you running?
 - v1.15.14+dev (39401aca02a-Modified)
 - Where did you download it from? (Steam, Mac App Store, our website, built from source, etc.)
 - Github
 - If you're using a custom build, what compiler did you use, and what commit did you build at?
 - v1.15.14+dev (39401aca02a-Modified) MSYS2
 - What OS are you running?
 - Windows 10

### Describe the bug
Game client segfaulted.

### To Reproduce
https://www.youtube.com/watch?v=1145nUmtXUo

### Expected behavior
I update the add-on and play it.

### Screenshots
![image](https://user-images.githubusercontent.com/5283677/124356430-c74daa80-dc37-11eb-8123-890832cb58c3.png)

### Additional context

I am not sure what caused it...but I can update every upgradeable add-on except Mercenary Band.

**Comment by Pentarctagon** [comment]
What version are you updating from?  The add-ons manager shows only 0.9 being available in the version selection dropdown.

**Comment by knyghtmare** [comment]
> What version are you updating from? The add-ons manager shows only 0.9 being available in the version selection dropdown.

I downloaded 0.8.1 earlier via the website and I was updating it to 0.9.0 and then segfault. I had a friend download the v0.9.0 and it segfaulted for him as well.

**Comment by Pentarctagon** [comment]
Hmmm.  I do wonder what's expected to happen in that scenario. @kabachuha or @irydacea what do you think?

I'm able to download 0.9.0 fresh without any issue, though.

**Comment by Wedge009** [comment]
Does #6389's resolution help here at all?

**Comment by Pentarctagon** [comment]
I haven't checked. Is it still reproducible?

**Comment by Pentarctagon** [comment]
@knyghtmare can you provide a backtrace? I'm still not able to reproduce this myself.

**Comment by knyghtmare** [comment]
I tried the procedure of reproducing this again but after the add-on ID mismatch was fixed by the server administrator, this default does not seem to occur anymore.

**Comment by knyghtmare** [comment]
Yes, that has been the case so far. Maybe there is some way to check the addon_id and use that as `dependencies=`
if incorrect, then the uploader will be prompted for a re-check?

**Comment by Pentarctagon** [comment]
A check on upload wouldn't quite cover all cases, since it would be possible to do something like:
* Upload the initial add-on and its dependency.
* Delete the dependency.
* Re-upload the dependency with letters in a different case than they were before.

**Comment by Pentarctagon** [comment]
I'm still not able to reproduce, unfortunately. I've uploaded the add-ons `test1` and `one Two three` to the 1.17 add-ons server, where `test1` lists its dependency as `one_two_three` rather than `one_Two_three` - can you try downloading `test1` and see if it crashes for you?

**Comment by knyghtmare** [comment]
Trying to download the test1 without having the dependency installed before gave this:
![image](https://user-images.githubusercontent.com/5283677/168414681-14fba32e-8c66-4934-a5e7-9784243eb152.png)

also, when I downloaded the dependency first, the test1 addon is showing the dependency as green and installed despite having reported it as broken before.

**Comment by Pentarctagon** [comment]
The only other option then I guess would be to try re-breaking the original add-on so you could get a stack trace - everything seems to work normally for me, and I don't know how to reproduce it with a different test add-on.

**Comment by Wedge009** [comment]
Is this still happening on 1.18? Noting the non-confirmed status.

**Comment by Wedge009** [comment]
Closing due to age. Re-open if it is still an issue.
